file.c

```diff
@@ -2618,3 +2616,83 @@ slist_t *file_parse_xmllike(char *name, char *tag)
   return sl0;
 }
 
+
+/*
+ * Very limited xml parsing.
+ *
+ * Returns a list of elements; cf. file_parse_xmllike_buf().
+ *
+ * name: name of file to read
+ * tag: elements name to look for
+ */
+slist_t *file_parse_xmllike(char *name, char *tag)
+{
+  slist_t *sl0 = NULL;
+  FILE *f;
+  char *buf = NULL;
+  int buf_size = 0, buf_ptr = 0, i;
+
+  if(!tag) return sl0;
+
+  if(!(f = fopen(name, "r"))) return sl0;
+
+  do {
+    buf = realloc(buf, buf_size += 0x1000);
+    i = fread(buf + buf_ptr, 1, buf_size - buf_ptr - 1, f);
+    buf_ptr += i;
+  }
+  while(buf_ptr == buf_size - 1);
+
+  buf[buf_ptr] = 0;
+
+  fclose(f);
+
+  if(!(buf_size = buf_ptr)) return sl0;
```

Comment:
Good catch.

(a) yes, '=' was intended.
(b) after the splitting of the function it's no longer needed at all